crates/ra_ide_api/src/lib.rs

```diff
@@ -396,6 +398,11 @@ impl Analysis {
         file_structure(&parse.tree())
     }
 
+    /// Returns a list of the places in the file where type hints can be displayed.
+    pub fn inlay_hints(&self, file_id: FileId) -> Vec<InlayHint> {
+        inlay_hints::inlay_hints(&self.db.parse(file_id).tree())
```

Comment:
This shoul use `.with_db` and return a `Cancellable`. Really, all methods here should be doing this, some don't but that's a bug

Reply:
Oh, ok.
Fixed for my case, should I take care of the others also?

Reply:
Yeah, we should fix this for other cases as well, but in a separate PR